src/llama_stack/providers/remote/vector_io/elasticsearch/elasticsearch.py

```diff
@@ -0,0 +1,364 @@
+            raise
+
+        return await self._results_to_chunks(results)
+
+    async def query_keyword(self, query_string: str, k: int, score_threshold: float) -> QueryChunksResponse:
+        """Keyword search using match query."""
+
+        try:
+            results = await self.client.search(
+                index=self.collection_name,
+                query={"match": {"chunk_content": {"query": query_string}}},
+                min_score=score_threshold,
+                limit=k,
+            )
+        except Exception as e:
+            log.error(f"Error performing keyword query on Elasticsearch index {self.collection_name}: {e}")
+            raise
+
+        return await self._results_to_chunks(results)
+
+    async def query_hybrid(
+        self,
+        embedding: NDArray,
+        query_string: str,
+        k: int,
+        score_threshold: float,
+        reranker_type: str,
+        reranker_params: dict[str, Any] | None = None,
+    ) -> QueryChunksResponse:
+        supported_retrievers = ["rrf", "linear"]
+        if reranker_type not in supported_retrievers:
+            raise ValueError(f"Unsupported reranker type: {reranker_type}. Supported types are: {supported_retrievers}")
+
+        retriever = {
+            reranker_type: {
+                "retrievers": [
+                    {"retriever": {"standard": {"query": {"match": {"chunk_content": query_string}}}}},
+                    {"retriever": {"knn": {"field": "vector", "query_vector": embedding.tolist(), "k": k}}},
+                ]
+            }
+        }
+
+        # Add reranker parameters if provided for RRF (e.g. rank_constant, rank_window_size, filter)
+        # see https://www.elastic.co/docs/reference/elasticsearch/rest-apis/retrievers/rrf-retriever
+        if reranker_type == "rrf" and reranker_params is not None:
+            allowed_rrf_params = {"rank_constant", "rank_windows_size", "filter"}
+            extra_keys = set(reranker_params.keys()) - allowed_rrf_params
+            if extra_keys:
+                raise ValueError(
+                    f"Unsupported RRF reranker parameters: {extra_keys}. Allowed parameters are: {allowed_rrf_params}"
+                )
+            retriever["rrf"].update(reranker_params)
+        elif reranker_type == "linear" and reranker_params is not None:
+            # Add reranker parameters (i.e. weights) for linear
+            # see https://www.elastic.co/docs/reference/elasticsearch/rest-apis/retrievers/linear-retriever
+            if await self._is_rerank_linear_param_valid(reranker_params) is False:
+                raise ValueError(
+                    "Invalid linear reranker parameters. Expected structure: "
+                    '{ "retrievers": { "standard": {"weight": float}, "knn": {"weight": float} } }'
+                )
+            # The first retriever is standard (see retriever variable above)
```

Comment:
the two dictionary element extractions are very brittle, i would suggest adding an explicit try and raise to explicitly indicate where the error occurs as this code feels like it could be bug ridden, even if it's expected behvaior